src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -318,11 +319,15 @@ def update_existing_check_in(
         and updated_status in CheckInStatus.USER_TERMINAL_VALUES
     )
 
-    if already_user_complete and not updated_duration_only:
-        # If we receive an in-progress check-in after a user-terminal value it
-        # could likely be due to the user's job running very quickly and events
-        # coming in slightly out of order. We can just ignore this type of
-        # error, and also return to not update the duration
+    # If we receive an in-progress check-in after a user-terminal value it
+    # could likely be due to the user's job running very quickly and events
+    # coming in slightly out of order. In this case we want to
+    # and also return to not update the duration
+    is_out_of_order_in_progress = (
+        already_user_complete and updated_status == CheckInStatus.IN_PROGRESS
+    )
```

Comment:
Should we log stats for this case? Is it useful to know if we're seeing more of these?